src/Connector.sol

```diff
@@ -10,16 +10,29 @@ interface GatewayLike {
     function transferTrancheTokensToCentrifuge(
         uint64 poolId,
         bytes16 trancheId,
+        address sender,
         bytes32 destinationAddress,
         uint128 amount
     ) external;
     function transferTrancheTokensToEVM(
         uint64 poolId,
         bytes16 trancheId,
-        uint256 destinationChainId,
+        address sender,
+        uint64 destinationChainId,
         address destinationAddress,
         uint128 amount
     ) external;
+    function transfer(uint128 currency, address sender, bytes32 recipient, uint128 amount) external;
+    function increaseInvestOrder(uint64 poolId, bytes16 trancheId, address investor, uint128 currency, uint128 amount)
+        external;
+    function decreaseInvestOrder(uint64 poolId, bytes16 trancheId, address investor, uint128 currency, uint128 amount)
+        external;
+    function increaseRedeemOrder(uint64 poolId, bytes16 trancheId, address investor, uint128 currency, uint128 amount)
+        external;
+    function decreaseRedeemOrder(uint64 poolId, bytes16 trancheId, address investor, uint128 currency, uint128 amount)
+        external;
+    function collectRedeem(uint64 poolId, bytes16 trancheId, address caller) external;
+    function collectInvest(uint64 poolId, bytes16 trancheId, address caller) external;
 }
 
```

Comment:
I can't leave a comment on that line, but the `Pool` struct still contains a `currency` key that afaik can be removed.

Reply:
Done, thanks for the catch